flycheck.el

```diff
@@ -6417,13 +6417,13 @@ about dogma."
     (nreverse errors)))
 
 (defun flycheck-elixir--check-for-dogma ()
-  "Check if the `dogma' subcommand is available.
+  "Check if `dogma' is installed and the file is saved.
 
-Check by parsing the output of `mix help'.
+Check by looking for deps/dogma.
 Used as a predicate for enabling the checker."
-  (let ((mix (flycheck-checker-executable 'elixir-dogma)))
-    (seq-find (lambda (l) (string-match-p "dogma" l))
-              (ignore-errors (process-lines mix)))))
+  (and
+   (flycheck-buffer-saved-p)
```

Comment:
Thanks, doesn't look too different, though.  I prefer the JSON parsing as I've more trust in structured formats and the other differences seem minor.

I think we'll just merge this PR 👍